packages/client/modules/userDashboard/components/OrgBilling/UpdatePayment.tsx

```diff
@@ -33,14 +50,119 @@ const CARD_ELEMENT_OPTIONS = {
 }
 
 type Props = {
-  handleCancel: () => void
+  handleClose: () => void
+  orgId: string
 }
 
 const UpdatePayment = (props: Props) => {
-  const {handleCancel} = props
+  const {handleClose, orgId} = props
+  const atmosphere = useAtmosphere()
+  const {onError, onCompleted} = useMutationProps()
+  const [isLoading, setIsLoading] = useState(false)
+  const stripe = useStripe()
+  const elements = useElements()
+  const [errorMsg, setErrorMsg] = useState<null | string>()
+  const [hasStarted, setHasStarted] = useState(false)
+  const [cardNumberError, setCardNumberError] = useState<null | string>()
+  const [expiryDateError, setExpiryDateError] = useState<null | string>()
+  const [cvcError, setCvcError] = useState<null | string>()
+  const [cardNumberComplete, setCardNumberComplete] = useState(false)
+  const [expiryDateComplete, setExpiryDateComplete] = useState(false)
+  const [cvcComplete, setCvcComplete] = useState(false)
+  const hasValidCCDetails =
```

Comment:
+1 Some form helper would be useful here. I think we have `useForm` hook. What do you think?

Reply:
From what I can see, `useForm` is used for holding the state of the strings. This is checking if the inputs are complete so I think regular `useState` is fine here